TACC/Core-CMS-Custom
# fix(templatetags): limit menu visibility + missing page id key = crash

## Overview

**Fix** that **if** any menu item does not have a `reverse_id`*†, **then** `limit_visibility_in_menu.py` will **crash**.

<sub>If a CMS page has no ID* set, it will still have a `reverse_id`, it's just blank. This fixes edge case of menu item **not having the attr/key**.†<sub>

<sub>* I.e. an "ID" in Advanced Settings.\
† E.g. CMS News category or article.</sub>

## Related

- copied from https://github.com/TACC/Core-CMS/pull/571
- adds safety to https://github.com/TACC/Core-CMS-Custom/pull/45

## Changes

- check for `reverse_id` before using it

## Testing

0. Load this branch on existing APCD setup.
1. Open any page of the site (not the CMS Administration).
2. Verify no crash (from newly introduced code).
3. Verify any of the test cases from [#45](https://github.com/TACC/Core-CMS-Custom/pull/45) still work.

## UI

N/A
